### DISCUSSION TITLE
How to set an event function to let next.js execute at the server starts?

### DISCUSSION THREAD

**Discussion by PrinOrange** [vercel/next.js#63502] [Help]
I have a search engine set up for my website that searches based on an in-memory index.
Therefore, before each search, the index must be built. But building the index takes a long time. So, I want to build the index every time the server starts, rather than every time the user calls the search API.
So, I am looking for a way. Does Next.js have a special API to let the specified function be executed when run the `next start` and server starts?
I once thought about executing it in getServerSideProps, but this would be repeated every time a page is requested, which is very inefficient.

Moreover, this executed function needs to run on the node server and call the node API, so it cannot be run in `useEffect` in the App component defined in the `_app.ts` file.

**Comment by icyJoseph** [comment]
Maybe give this a go? https://nextjs.org/docs/app/building-your-application/optimizing/instrumentation Read the entire document though.
